### PR TITLE
Hide the breadcrumb menu if the node is not within the given path

### LINKED ISSUE #856: eval.path does not work correctly with the new fileTree picker
**Affected version(s)** 4.8 (probably also 4.4?)

If I go to the file manager and select the folder *files/foo*, this folder gets stored in the session under `tl_files_node`.

If I then click on a `fileTree` widget which has the configuration `eval => ['path' => 'files/bar']` the breadcrumbs incorrectly display *files/foo* and no file or folder is visible:

Screenshot_2019-10-17 Sale › Produkt ID 374 bearbeiten area at 127 0 0 1 xip io

The problem is that the breadcrumbs logic gets executed `onload_callback` https://github.com/contao/contao/blob/4.8/core-bundle/src/Resources/contao/classes/Backend.php#L1046 and the picker only afterwards https://github.com/contao/contao/blob/4.8/core-bundle/src/Resources/contao/drivers/DC_Folder.php#L3073-L3074

The old file selector has the logic for this issue here: https://github.com/contao/contao/blob/4.3.0/core-bundle/src/Resources/contao/widgets/FileSelector.php#L211-L226

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #856

@ausi I am not 100% sure if this is the best place to fix the issue. Can you check as well please?

### CODE REVIEW

File: core-bundle/src/Resources/contao/drivers/DC_Folder.php [hunk 1]
```diff
@@ -3003,6 +3003,14 @@ public function initPicker(PickerInterface $picker)
 				throw new \RuntimeException('Invalid path ' . $strPath);
 			}
 
+			$strNode = System::getContainer()->get('session')->getBag('contao_backend')->get('tl_files_node');
+
+			// If the files node is not within the current path, remove it (see #856)
+			if ($strNode && ($i = array_search($strNode, $this->arrFilemounts)) !== false && strncmp($strNode . '/', $strPath . '/', strlen($strPath) + 1) !== 0)
```
Comment (m-vo) [inline core-bundle/src/Resources/contao/drivers/DC_Folder.php:3009]: IMHO those long conditions + assignments are way more readable in yoda style + linebreaks between the topmost parts:

```php
if ($strNode
    && false !== ($i = array_search($strNode, $this->arrFilemounts))
    && 0 !== strncmp($strNode . '/', $strPath . '/', strlen($strPath) + 1)
)
```
but then again it's DC_Folder :grinning:

**Comment by ausi** [comment]
This fix seems to work partially. 🙁 
After clicking “Change selection” it first shows the correct directory, but if I then click on the “Upload files” button the folder switches back to the one that is stored in the session.

*(Tested in Contao 4.8 with the changes of this PR manually applied)*

**Comment by leofeyer** [comment]
> After clicking “Change selection” it first shows the correct directory, but if I then click on the “Upload files” button the folder switches back to the one that is stored in the session.

This was caused by the line of code that I have removed in 492b74b0c27145801484f7a8e4e6a75b13403a20. Can you please check again?

**Comment by ausi** [comment]
Yes, works 🎉

**Review by ausi — APPROVED** [review]
I don’t fully understand how this works 🙈 But it fixes the issue 🙃

**Pull request merged**